bitswap.go

```diff
@@ -154,8 +154,15 @@ func WithTargetMessageSize(tms int) Option {
 	}
 }
 
+func WithPeerBlockRequestFilter(pbrf PeerBlockRequestFilter) Option {
+	return func(bs *Bitswap) {
+		bs.peerBlockRequestFilter = pbrf
+	}
+}
```

Comment:
(opening this here so we get a discussion thread)

There are other requests for [bitswap configuratibility](https://github.com/ipfs/go-bitswap/issues/532) and I guess we'll want to make these options reusable accross the stack (graphsync for example).

We've only implemented the obvious predicate, but what if we reused a pattern like the [ConnectionGater](https://github.com/libp2p/go-libp2p-core/blob/master/connmgr/gater.go)?
Something like:

```go
type ContentRoutingGater interface {
   InterceptBlockRequest(peerId, blockId) (allow bool)
   PrioritizeBlockRequest(peerA, peerB) (order int)
}
```

Is it too early, or would it make sense to give it some more design thought?

Reply:
I think it's fair to pass an interface rather than a function, but I don't think it's worth grouping all of the options into a single interface (i.e. you could have an interface with one function). Otherwise you end up with a bunch of implementations that have "error not supported" for various functions instead of using interface assertions to figure that out.

At the moment the task comparator is a function rather than an interface so let's just stick with using a function here for now and we can always change it later